source/common/router/scoped_rds.cc

```diff
@@ -263,25 +253,29 @@ void ScopedRdsConfigSubscription::onConfigUpdate(
         });
   }
 
-  std::vector<std::string> exception_msgs;
+  std::string exception_msg;
+  Protobuf::RepeatedPtrField<std::string> clean_removed_resources;
+  detectUpdateConflictAndCleanupRemoved(added_resources, removed_resources, clean_removed_resources,
```

Comment:
nit: it's generally preferable for readability to use a return arg as opposed to output parameters.

Reply:
Agreed, I have changed clean_removed_resources to be the explicitly returned value.